DSA Session/Assignment4/Session4/src/Question1/Employee.java

```diff
@@ -0,0 +1,85 @@
+package Question1;
+
+/**
+ * Represents Employee.
+ * 
+ * @author Digvijay
+ *
+ */
+public class Employee implements Comparable<Employee> {
+
+    private int id;
+    private String employeeName;
+    private String employeeAddress;
+
+    /**
+     * Constructs the employee with the given id, name and address.
+     * @param id : id of the employee
+     * @param employeeName : name of the employee
+     * @param employeeAddress : address of the employee
+     * @throws Exception if any of the field in invalid.
+     */
+    public Employee(int id, String employeeName, String employeeAddress)
+            throws Exception {
+
+        if (id < 0) {
+            throw new Exception("Invalid employee id");
+        }
+        if (employeeName.trim().length() == 0) {
```

Comment:
http://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html

Take a look there are StringUtil methods this can be used to check null and length type things.